tests/clock/ClockUtils.py

```diff
@@ -0,0 +1,384 @@
+import logging
+import random
+import datetime
+import allure
+import time
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.clock.ClockConsts import ClockConsts
+from tests.common.errors import RunAnsibleModuleFail
+
+
+class ClockUtils:
+    @staticmethod
+    def run_cmd(duthosts, cmd, param=''):
+        """
+        @summary:
+            Run a given command and return it's output.
+            * a successful command returns empty output (''), while failure returns error message
+        @return: commands output (str)
+        """
+        with allure.step('Run command: "{}" with param "{}"'.format(cmd, param)):
+            logging.info('Run command: "{}" with param "{}"'.format(cmd, param))
+            DUT_HOSTNAME = duthosts[0].hostname
+
+            cmd_to_run = cmd if param == '' else cmd + ' ' + param
+            logging.info('Actual command to run: "{}"'.format(cmd_to_run))
+
+            try:
+                cmd_output = duthosts.command(cmd_to_run)[DUT_HOSTNAME][ClockConsts.STDOUT]
+            except RunAnsibleModuleFail as cmd_err:
+                output = cmd_err.results[ClockConsts.STDOUT]
+                err = cmd_err.results[ClockConsts.STDERR]
+                cmd_output = output if output else err
+                logging.info('Command Error!\nError message: "{}"'.format(cmd_output))
+            logging.info('Output type: {}'.format(type(cmd_output)))
+            logging.info('Output: {}'.format(cmd_output))
+
```

Comment:
Does it mean that all the errors will be suppressed?

Reply:
according to my manual experiments with the commands, when a command succeeds, it's output is only in stdout, and when something fails it can output to stdout or stderr, but not both, so i take the non-empty one.

Reply:
@roy-sror can resolve conversation?